README.md

```diff
@@ -158,6 +158,39 @@ And here's an example of how you might document a class:
      ) { }
 ```
 
+Puppet providers often rely on different types of metaprogramming to create
+parameters and methods. Since Strings does not execute the code which is being
+documented you must provide hints on how to document your code. To document a
+parameter which is automatically created you must use the special directive
+`@!puppet.provider.param` Which may take types, the parameter name, and a
+description.
+
+Strings will automatically extract the `@doc` provider docstring and any `desc`
+parameter docstrings.
+
+```ruby
+# @!puppet.provider.param my_parameter This parameter needs to be explicitly
```

Comment:
1. Your wordsmithing is very nice. Would you like top open a PR against the README?
2. I didn't know you could create a property automatically. We should probably add a directive to support that. If you don't mind, could you file a bug in JIRA? If you're really curious about Strings internals I could give you some pointers on how to implement a directive.

Thanks for your help!